config/opal_check_libnl.m4

```diff
@@ -0,0 +1,223 @@
+                              AS_IF([echo $ldd_output | grep -q libnl.so],
+                                    [AS_IF([test $opal_libnl_version -eq 3],
+                                           [AC_MSG_WARN([lib nl version conflict: $opal_libnl3_libs requires libnl-3 whereas $1 requires libnl])],
+                                           [opal_libnl_libs="$opal_libnl_libs $1"
+                                            opal_libnl_version=1])
+                                     libnl_version=1])
+                              AS_IF([echo $ldd_output | grep -q libnl-3.so],
+                                    [AS_IF([test $libnl_version -eq 1],
+                                           [AC_MSG_ERROR([lib $1 requires both libnl and libnl3\nthis is just insane\nAborting...])])
+                                     AS_IF([test $opal_libnl_version -eq 1],
+                                           [AC_MSG_WARN([lib nl version conflict: $opal_libnl_libs requires libnl whereas $1 requires libnl-3])],
+                                           [opal_libnl3_libs="$opal_libnl3_libs $1"
+                                            opal_libnl_version=3])])
+                              rm -f conftest conftest_c.$ac_ext],
+                             [AC_MSG_WARN([Could not link a simple program with lib $1])])
+            AC_LANG_POP(C)
+            ;;
+        esac
+])
+
+dnl
+dnl Check for libnl-3.
+dnl
+dnl Inputs:
+dnl
+dnl $1: prefix where to look for libnl-3
+dnl $2: var name prefix of _CPPFLAGS and _LDFLAGS and _LIBS
+dnl
+dnl Outputs:
+dnl
+dnl - Set $2_CPPFLAGS necessary to compile with libnl-3
+dnl - Set $2_LDFLAGS necessary to link with libnl-3
+dnl - Set $2_LIBS necessary to link with libnl-3
+dnl - Set HAVE_LIBNL3 1 if libnl-3 will be used
+dnl
+AC_DEFUN([OPAL_CHECK_LIBNL3],[
+    OPAL_VAR_SCOPE_PUSH([CPPFLAGS_save LIBS_save LDFLAGS_save])
+    AC_MSG_NOTICE([checking for libnl3])
+
+    AC_MSG_CHECKING([for libnl3 prefix])
+    AC_MSG_RESULT([$1])
+    AC_MSG_CHECKING([for $1/include/libnl3])
+    AS_IF([test -d "$1/include/libnl3"],
+          [opal_libnl3_happy=1
+           AC_MSG_RESULT([found])],
+          [opal_libnl3_happy=0
+           AC_MSG_RESULT([not found])])
+
+    # Random note: netlink/version.h is only in libnl3 - it is not in libnl.
+    # Also, nl_recvmsgs_report is only in libnl3.
+    CPPFLAGS_save=$CPPFLAGS
+    opal_tmp_CPPFLAGS="-I$1/include/libnl3"
+    CPPFLAGS="$opal_tmp_CPPFLAGS $CPPFLAGS"
+    AS_IF([test $opal_libnl3_happy -eq 1],
+          [OPAL_CHECK_PACKAGE([$2],
+                              [netlink/version.h],
+                              [nl-3],
+                              [nl_recvmsgs_report],
+                              [],
+                              [$1],
+                              [],
+                              [opal_libnl3_happy=1],
+                              [opal_libnl3_happy=0])
+
+           # Note that FI_CHECK_PACKAGE is going to add
```

Comment:
Minor copy-n-paste error here: should be OPAL_CHECK_PACKAGE, not FI_CHECK_PACKAGE.